### ISSUE TITLE
RT shadow upscaling quality

### ISSUE THREAD

**Issue by trsh** [turanszkij/WickedEngine#613]
Recently I posted an issue with vertical CRT effect & blurriness with FSR2. There is a fix now that makes the scene so much clearer and usable (as it should be). But I think the vertical CRT's is different problem. As matter of fact I get them with no upscale with just RT shadow on. Shadow maps don't have this problem.

![2022-12-24_103637](https://user-images.githubusercontent.com/3727523/209428005-2717c461-2abc-4bc7-b82b-042e54633425.png)

**Comment by trsh** [comment]
Also when moving around with RT shadows, lots of blinking and pixel moving on edges and beyond

![2022-12-24_104253](https://user-images.githubusercontent.com/3727523/209428164-413ae388-46ef-4715-a9fe-e473d319b90a.png)

**Comment by turanszkij** [comment]
Yes, this is because RT shadows are running on half resolution, and the upscaling for it is not good.

**Comment by turanszkij** [comment]
Commit https://github.com/turanszkij/WickedEngine/commit/906638c36950872523556d001f8e2e7d4bf645a0 adds an upscale pass to raytraced shadows. Not perfect, but slightly better.

I'm considering that most of these half res heavy effects would be better to always run in full resolution, and FSR2 upscaler to be used for the whole rendering (if needed) instead of having upscaling per effect.

**Comment by trsh** [comment]
Tnx, I test it as soon as I can

**Comment by trsh** [comment]
As u said. Slightly better. Maybe an option to use full res would make sense. 

![2023-01-07_165820](https://user-images.githubusercontent.com/3727523/211157013-1b5d7671-2b1f-4e00-96e9-78343242f7bc.png)
